src/dehtml.rs

```diff
@@ -126,22 +173,28 @@ fn dehtml_quick_xml(buf: &str) -> String {
         buf.clear();
     }
 
-    dehtml.strbuilder
+    (dehtml.strbuilder, dehtml.quote)
 }
 
 fn dehtml_text_cb(event: &BytesText, dehtml: &mut Dehtml) {
+    static LINE_RE: Lazy<regex::Regex> = Lazy::new(|| regex::Regex::new(r"(\r?\n)+").unwrap());
+
     if dehtml.get_add_text() == AddText::YesPreserveLineEnds
         || dehtml.get_add_text() == AddText::YesRemoveLineEnds
     {
         let last_added = escaper::decode_html_buf_sloppy(event as &[_]).unwrap_or_default();
 
         if dehtml.get_add_text() == AddText::YesRemoveLineEnds {
-            dehtml.strbuilder += LINE_RE.replace_all(&last_added, "\r").as_ref();
-        } else if !dehtml.line_prefix().is_empty() {
-            let l = dehtml.append_prefix("\n");
-            dehtml.strbuilder += LINE_RE.replace_all(&last_added, l.as_str()).as_ref();
+            let last_added = LINE_RE.replace_all(&last_added, " ");
```

Comment:
It took me quite a long time to understand this code, so I think we should add a comment:

```suggestion
            // We want to remove line ends.
            // - Replace all line ends with a space. \r\n\r\n is replaced with just one space.
            // - It can happen that `last_added` starts with whitespace, _and_ `buf` ends with
            //   a space. So, add exactly one space, if we don't have one yet, and strip the rest.
            let last_added = LINE_RE.replace_all(&last_added, " ");
```

Assuming I got this right.

Also, I noticed that if the cb tag's text starts with multiple spaces, they are replaced with just a single space. But if there are multiple consecutive spaces within, they are left as-is. Is this OK?

Also also (that's unrelated to this PR), `last_added` is confusing as the variable name, maybe rename it to `decoded_text`?